Library/Homebrew/dependency.rb

```diff
@@ -11,18 +11,20 @@ class Dependency
   include Dependable
   extend Cachable
 
-  attr_reader :name, :env_proc, :option_names
+  attr_reader :name, :env_proc, :option_names, :tap
 
   DEFAULT_ENV_PROC = proc {}.freeze
   private_constant :DEFAULT_ENV_PROC
 
-  def initialize(name, tags = [], env_proc = DEFAULT_ENV_PROC, option_names = [name])
+  def initialize(name, tags = [], env_proc = DEFAULT_ENV_PROC, option_names = [name&.split("/")&.last])
     raise ArgumentError, "Dependency must have a name!" unless name
 
     @name = name
     @tags = tags
     @env_proc = env_proc
     @option_names = option_names
+
+    @tap = Tap.fetch(name.rpartition("/").first) if name.match?(HOMEBREW_TAP_FORMULA_REGEX)
```

Comment:
Not blocking: given how often we do this all over the place: feels like we should have a better accessor for this sort of thing.

Reply:
Probably.

I've tweaked this slightly in the meantime to:

```rb
@tap = Tap.fetch(Regexp.last_match(1), Regexp.last_match(2)) if name =~ HOMEBREW_TAP_FORMULA_REGEX
```